lib/rules/func-name-matching.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * @fileoverview Rule to require function names to match the name of the variable or property to which they are assigned.
+ * @author Annie Zhang,
+ */
+
+"use strict";
+
+//--------------------------------------------------------------------------
+// Helpers
+//--------------------------------------------------------------------------
+
+/**
+ * Gets the property name from a pattern
+ * @param {ASTNode} pattern The left side of the AssignmentExpression
+ * @returns {string} The name of the variable or property being assigned, or an empty string if no valid name is found.
+ */
+function getName(pattern) {
+    switch (pattern.type) {
+
+        // prop = ...;
+        case "Identifier":
+            return pattern.name;
+
+        // obj["prop"] = ...;
+        case "Literal":
+            if (typeof (pattern.value) === "string") {
```

Comment:
It seems it is not common to use parentheses in eslint code base, might be an argument to enable `no-extra-parens`